spl/labels.py

```diff
@@ -186,55 +198,90 @@ def __get_application_numbers(self, set_id, bs_content):
         return list(application_numbers)
 
     def __get_label_text(self, set_id, bs_content):
-        def get_xml_text(navigable_title):
-            # Process the label text
-            label_text = (
-                re.sub(r"\n{3,}", "\n\n", navigable_title.get_text())
-                # Remove initial / trailing white space
-                .lstrip().rstrip()
-            )
-            return label_text
+        def get_xml_text(text):
+            # Process the label text; normalize remove nbsp
+            text = unicodedata.normalize("NFKC", text.lstrip().rstrip())
+            text = re.sub(r"\n\s*\n", "\n", text)
+            return text
+
+        def get_xml_title(text):
+            # strip extra \n, \t, and spaces in titles, but keep number
+            text = get_xml_text(text)
+            text = re.sub(r"\.\s+", ".", text)
+            text = re.sub(r"\s+", " ", text)
+            return text
 
         # Get and process all "title" tags
         titles = bs_content.find_all("title")
         labels = []
 
-        for title in titles:
-            if isinstance(title, Tag):
-                for section_title in SplHistoricalLabels.LABEL_SECTIONS:
-                    if section_title in title.text:
-                        # Match the title tags in the sub-section
-                        sub_titles = title.parent.find_all("title")
-                        sub_sections = []
-                        for sub_title in sub_titles:
-                            if isinstance(sub_title, Tag):
-                                for (
-                                    sub_section_title
-                                ) in SplHistoricalLabels.LABEL_SECTIONS[
-                                    section_title
-                                ]:
-                                    if sub_section_title in sub_title.text:
-                                        # Extract the sub-title text
-                                        sub_sections.append(
-                                            {
-                                                "name": sub_section_title,
-                                                "text": get_xml_text(
-                                                    sub_title.parent
-                                                ),
-                                            }
-                                        )
-                                        # Remove the sub title from the main title
-                                        sub_title.parent.decompose()
-
-                        # Add the section text, along with the sub-sections
+        i = 0
+        while i < len(titles):
```

Comment:
This section feels a little verbose, I think it can be refactored to extract some common logic into a function (eg: there is a single `labels.append()`)

Reply:
See comment above.